### DISCUSSION TITLE
in advance what the v2 features are and what the design considerations are?

### DISCUSSION THREAD

**Discussion by leeyisoft** [flyerhq/flutter_chat_ui#560] [Q&A] [answered]
@demchenkoalex  You lead the discussion. For now, I'm just an avid spectator.

https://github.com/flyerhq/flutter_chat_ui/issues/465#issuecomment-1947990758

currently v2 looks like not much, because I focused on architecture and performance first. As of right now I have:

* dynamic theming (changes based on a device theme, easier configuration)
* thumhash (blurred image) preview while image loads
* cross platform media cache (including web, windows, linux) so much better than https://pub.dev/packages/cached_network_image which is iOS/Android/Mac
* performance (renders 10s of thousands of messages, while v1 struggles with 50)
* architecture based on abstract classes, so it is ready for any databases people might want
* much easier messages animation handling

**Comment by demchenkoalex** [comment]
hey, I anyway planned to open a big issue with progress done and remaining features + close issues with checkboxes to fix in 2.0 + release an alpha 2.0 build, when at least sending messages work. So I think this discussion is too early right now, but you will definitely see a bigger issue soon. I am currently testing my new architecture with rest api / web sockets and firebase (among other tests to be added) and hopefully everything works perfectly and I will be able to do all this and also open 2.0 to contributions.

**Comment by demchenkoalex** [comment, marked as answer]
promised issue is here https://github.com/flyerhq/flutter_chat_ui/issues/646
